### PR TITLE
[receiver/purefa] Pure Storage FlashArray - Purity v6.6.11+ Native OpenMetrics Support

### PR DESCRIPTION
**Pull request by chrroberts-pure**
This PR implements the changes required to scrape metrics from arrays with newer Purity versions. 

In Pure Storage FlashArray Purity version 6.6.11, they released a feature that allows OpenMetrics to be scraped directly from the array without the use of an OpenMetrics Exporter. 

These changes include
- Enabling support to disable TLS Insecure Skip Verify if required (default is to verify)
- Enabling support for the `namespace` HTTP param that is supported by the Purity OpenMetrics exporter.

#### Link to tracking issue
None

#### Testing
Tests have been updated for the scraper and default configs for the new Namespace and TLSInsecureSkipVerify configurations. 

I have tested this with a live FlashArray running 6.6.11

#### Documentation

The Readme was updated to show a new example of a scrape using Purity version 6.6.11+.

### CODE REVIEW

**Comment by atoulme** [comment]
Please add tests.

**Comment by chrroberts-pure** [comment]
@atoulme Tests have been added. Thank you.

File: receiver/purefareceiver/README.md
Comment (jpkrohling) [inline receiver/purefareceiver/README.md:69]: I wonder if it's possible to obtain the server CA in some way, and use it here. Something for a follow-up PR though.

File: receiver/purefareceiver/README.md
Comment (jpkrohling) [inline receiver/purefareceiver/README.md:21]: It looks like the new "namespace" attribute is also required. It needs to be placed here.

File: receiver/purefareceiver/internal/scraper.go
Comment (jpkrohling) [inline receiver/purefareceiver/internal/scraper.go:80]: Interesting, what if people set the other options from tls config, are they going to be ignored? You could probably pass the entire TLSConfig from the config to the scraper, no? In fact, we could probably just reuse the entire client config here...

File: receiver/purefareceiver/README.md
Comment (chrroberts-pure) [reply receiver/purefareceiver/README.md:69]: I don't think is quite necessary, as if the end user wanted the connection to be trusted they would import the CA into their root certs on their system to trust the array.

File: receiver/purefareceiver/README.md
Comment (chrroberts-pure) [reply receiver/purefareceiver/README.md:21]: fixed.

File: receiver/purefareceiver/internal/scraper.go
Comment (chrroberts-pure) [reply receiver/purefareceiver/internal/scraper.go:80]: This may be out of my skill range to do that. I tried passing the reciever tlsconfig directly to the Prometheus TLSConfig, and they were incompatible.  I'll try and poke around some more for a cleaner way to do this. Please let me know if there's any hints you can give me. thank you!

Comment (jpkrohling) [reply receiver/purefareceiver/internal/scraper.go:80]: You are right -- they are similar, but of the same type. Perhaps something like:

```go
type scraper struct {
	scraperType    ScraperType
	endpoint       string
	namespace      string
	tlsSettings    configtls.ClientConfig
	configs        []ScraperConfig
	scrapeInterval time.Duration
	labels         model.LabelSet
}
...
func (h *scraper) ToPrometheusReceiverConfig(host component.Host, _ receiver.Factory) ([]*config.ScrapeConfig, error) {
...
		httpConfig.TLSConfig = configutil.TLSConfig{
			InsecureSkipVerify: h.tlsSettings.InsecureSkipVerify,
			ServerName:         h.tlsSettings.ServerName,
			CAFile:             h.tlsSettings.CAFile,
			CertFile:           h.tlsSettings.CertFile,
			KeyFile:            h.tlsSettings.KeyFile,
			...
		}
...
}
```

The goal is to provide a way that is consistent across components, while setting the right expectations to users. In your case, the UX is consistent, as Collector users are exposed to the same config used in other components, but if they set `MinVersion`, this isn't going to be respected.

I'd recommend to either log a warning if anything else other than the `InsecureSkipVerify` flag is set, or to support the most options possible (which include more than what I added there explicitly).

**Review by jpkrohling — APPROVED** [review]

File: receiver/purefareceiver/internal/scraper.go
Comment (chrroberts-pure) [reply receiver/purefareceiver/internal/scraper.go]: Some of these will be harder to implement than others - eg. MinVersion for the prom http config is a uint16, I've otherwise mapped the rest of the struts. 

Let me know what you think, and i'll also test with an array today.

**Comment by chrroberts-pure** [comment]
Hi @jpkrohling @atoulme I'm happy with this, and I've tested with a Pure FlashArray. Please let me know if there is anything else that is blocking the merge of this PR. Thank you.

**Review by jpkrohling — APPROVED** [review]
LGTM, the codeowners check failure can be ignored. Once the conflict is sorted out, this can be merged.

**Review by jpkrohling — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Pull request merged**
